cli/src/main/java/org/eclipse/hono/cli/client/Client.java

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.cli.client;
+
+import org.eclipse.hono.cli.AbstractCliClient;
+import org.eclipse.hono.config.ClientConfigProperties;
+import org.springframework.shell.Availability;
+import org.springframework.shell.standard.ShellComponent;
+import org.springframework.shell.standard.ShellMethod;
+import org.springframework.shell.standard.ShellMethodAvailability;
+import org.springframework.shell.standard.ShellOption;
+import org.springframework.stereotype.Component;
+
+/**
+ * This class contains the methods that the user can execute to create or destroy the factory given connection info.
+ * The vertx instance wil be prepared and the factory correspondent to the profile chosen will be created.
+ */
+@ShellComponent
+@Component
+public class Client extends AbstractCliClient {
+
+    @ShellMethod("Create the connection to a given endpoint\n" +
+            "\t\t\t[-H --host] [-P --port] [-u --username] [-p --password]")
+    void connect(@ShellOption(value = {"-H", "--host"},  defaultValue = "", help = "The address of the endpoint which you want to establish a connection to") final String host,
```

Comment:
this belongs to the *ApplicationProfile* and the *DeviceProfile* FMPOV

Reply:
I disagree on that:
The user asks to connect submitting parameters [-H --host] [-P --port] [-u --username] [-p --password]
Then he will be prompted to insert the profile:

> Profile:
> [APPLICATION]: Connect to the Messaging Network
> [DEVICE]: Connect to the AMQP Adapter
> Please enter one of the [] values

(*)*There the factory corresponding to the profile will be created with the connection instance, but still not connected*

And return a positive feedback to the user that can now use the methods of the profile choosen. 

I thought that you asked for something like that. But say me if i'm wrong.

Reply:
If this is all right i have a small issue with the AMQP factory. @b-abel (Sorry to throw you in the middle of the conversation i'll try to give you a bit of context)
If the behaviour in the previusly message is right, at the moment (*), when i create the factory (that can be a factory with the application methods or a factory with device methods to interact with the AMQP adapter) i don't have the tenantId that i need to create the AMQP factory.

Do i need to ask the user for the tenantId previusly?
Or maybe it's my code that is bad structured?
Or i need to specify the tenant later adding a method in the factoryImpl to set it?

Reply:
> I thought that you asked for something like that. But say me if i'm wrong.


I think that we want to achieve the same thing. My proposal is simply to have specific methods for connecting to the endpoints
* *connectApplication*
* *connectAmqp*

Later we will probably also want to add *connectMqtt* in order to connect to the MQTT adapter ...
Using specific methods allows us to increase cohesion within the **Profile* classes FMPOV.

Reply:
ok i thought that a single connect command would be nicer but with future implementation maybe it's less practical.

So in the connectAmqp do i need to acquire also the tenantId?

Reply:
the tenant ID is usually part of the username: `{auth-id}@{tenant-id}` so no need to have an extra param for that FMPOV